driver/pparse.ml

```diff
@@ -183,7 +189,7 @@ let file_aux ~tool_name inputfile (type a) parse_fun invariant_fun
   let ast =
     Profile.record_call "-ppx" (fun () ->
       apply_rewriters ~restore:false ~tool_name kind ast) in
-  if is_ast_file || !Clflags.all_ppx <> [] then invariant_fun ast;
+  if is_ast_file && !Clflags.all_ppx = [] then invariant_fun ast;
```

Comment:
I find the logic for this line here fairly obscure. Could you just move the `invariant_fun` call to within the `is_ast` branch, so that it would test the invariants before the ppx extensions are run on the file? (This is assuming that checking the invariants is cheap, so that doing it twice is not a big deal.)

Reply:
(Or if we don't want to check both before and after rewriting, you can still condition on `all_ppx = []` within the branch.)

Reply:
Maybe adding a comment on the logic and moving it to the `is_ast` branch is sufficient?
```OCaml
        let ast = (input_value ic : a) in
        if !Cflags.all_ppx = [] then invariant_fun ast;
        (* if all_ppx <> [], invariant_fun will be called by apply_rewriters *)
        ast
```

Reply:
Yes, it seems reasonable to me.

Reply:
Done.